### PR TITLE
Use helpers.sh for waitForPod function and add early exit on `ImagePullBackoff`

### PR DESCRIPTION
**Pull request by SimonBaeumer**
- Move `helpers.sh` to `dev/lib/helpers.sh`
 - Use `waitForPod` from centralized location in all start scripts
 - Exit early if an ImagePullBackoff is detected

### CODE REVIEW

File: dev/lib/helpers.sh [hunk 1]
```diff
@@ -31,31 +31,35 @@ function waitForPod() {
     MINUTE=0
     podName=$1
     ignore=$2
-    running="\([0-9]\+\)\/\1"
-    printf "${BLUE}Waiting for ${podName} to reach running state (4min).\n${CLEAR}"
```
Comment (SimonBaeumer) [inline dev/lib/helpers.sh:35]: Removed the colours as they weren't used in `start.sh` nor in `dev/start.sh`.
Please double-check if this change is okay.

Comment (gurnben) [reply dev/lib/helpers.sh:35]: I actually added the coloration (hopefully with a toggle to turn them off) in brew-start.sh in hopes that we would get some cycles to properly re-touch start.sh and dev/start.sh to clean them up a bit.  Admittedly coloration isn't the cleanest thing - so I'm _definitely_ open to feedback - do you find it disruptive/obtrusive.  I'm not opposed to wholesale removing it.  

I also now see that you've done the heavy lifting of removing the baked-in versions of the helpers files I broke out (but importantly didn't change references - we were late in the release and didn't want to risk destabilizing CI).

File: brew-start.sh [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 
 #-----BOOTSTRAP HELPERS-----#
-source helpers.sh
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (gurnben) [inline brew-start.sh:10]: _Nice trick!_ - I'm definitely going to steal this 😄

Comment (gurnben) [reply brew-start.sh:10]: Our other solution uses some bits that don't work on MacOS and we didn't want to break MacOS compatibility but this should be fine!

File: dev/lib/helpers.sh (hunk 1 above)
Comment (gurnben) [reply dev/lib/helpers.sh:35]: Thanks!

**Review by gurnben — APPROVED** [review]
LGTM!

**Pull request merged**